smarts/core/sumo_road_network.py

```diff
@@ -171,32 +162,47 @@ def from_file(
                 # coordinates are relative to the origin).
                 G._shifted_by_smarts = True
 
-        return cls(
-            G,
-            net_file,
-            default_lane_width=default_lane_width,
-            lanepoint_spacing=lanepoint_spacing,
-        )
+        return cls(G, net_file, map_spec)
 
     @property
     def source(self) -> str:
         """This is the net.xml file that corresponds with our possibly-offset coordinates."""
         return self._net_file
 
-    def is_same_map(self, map_spec: MapSpec) -> bool:
-        dlw = (
+    @staticmethod
+    def _spec_lane_width(map_spec: MapSpec) -> float:
+        return (
             map_spec.default_lane_width
             if map_spec.default_lane_width is not None
             else SumoRoadNetwork.DEFAULT_LANE_WIDTH
         )
-        return (
-            map_spec.source == self._net_file
-            and (
-                (not map_spec.lanepoint_spacing and not self._lanepoints)
-                or map_spec.lanepoint_spacing == self._lanepoints.spacing
+
+    def is_same_map(self, map_spec: MapSpec) -> bool:
+        if map_spec.source != self._map_spec.source:
+            cur_source = (
+                map_spec.source
+                if os.path.isfile(map_spec.source)
+                else os.path.join(map_spec.source, "map.net.xml")
             )
-            and dlw == self._default_lane_width
-        )
+            orig_source = (
+                self._map_spec.source
+                if os.path.isfile(self._map_spec.source)
+                else os.path.join(self._map_spec.source, "map.net.xml")
+            )
+            if cur_source != orig_source:
+                return False
+
+        if map_spec.lanepoint_spacing != self._map_spec.lanepoint_spacing:
+            return False
+
+        if (
+            map_spec.default_lane_width != self._map_spec.default_lane_width
+            and SumoRoadNetwork._spec_lane_width(map_spec)
+            != SumoRoadNetwork._spec_lane_width(self._map_spec)
```

Comment:
Isn't this check redundant since it is being checked by the first part of this and statement?

Reply:
The second part here is to catch the case where the original `map_spec` specified `None` whereas the new one explicitly specifies `3.2` (which is what `DEFAULT_LANE_WIDTH` is).  In that case, we'd still consider these the "same map" even though their `map_specs` are not identical.